### PR TITLE
Redesign of footer to include new links and design

### PR DESCRIPTION
**Pull request by spencerldixon**
### Trello card

https://trello.com/c/basEB8lp/6758-build-new-footer-to-include-contact-centre-phone-and-webchat-plus-links-for-seo

### Context

We want to change our footer so that:
- it includes the contact centre helpline and webchat details
- it includes links to key pages for SEO

### Changes proposed in this pull request

- Refactor footer to include "Talk to us" section
- Redesign footer from dark theme to light theme
- Remove unused components

### CODE REVIEW

File: app/components/footer_component.html.erb [hunk 1]
```diff
@@ -1,35 +1,98 @@
...
+              <h3 class="heading-s">Chat to us</h3>
+              <%= render ChatComponent.new %>
+            </div>
+          </div>
+        <% end %>
       </div>
+
+      <div>
+        <p>Call or chat to us Monday to Friday, <strong>8:30am until 5:30pm</strong> except on <a class="link" href="https://www.gov.uk/bank-holidays">bank holidays</a>.</p>
+        <p class="small">Your details are protected under the terms of our <%= link_to "privacy notice", privacy_policy_path, class: "link" %> . Our privacy notice explains how we use your personal data. It's important you read this before chatting to us.</p>
```
Comment (gemmadallmandfe) [inline app/components/footer_component.html.erb:30]:
```suggestion
        <p class="small">Your details are protected under the terms of our <%= link_to "privacy notice", privacy_policy_path, class: "link" %>. Our privacy notice explains how we use your personal data. It's important you read this before chatting to us.</p>
```

File: app/components/footer_component.html.erb [hunk 2]
```diff
@@ -1,35 +1,98 @@
...
+      </div>
+    </section>
+
+    <section>
+      <hr class="site-footer__divider">
```
Comment (gemmadallmandfe) [inline app/components/footer_component.html.erb:35]: This (and the subsequent `<hr>` is read out by VoiceOver as "separator"

I have read mixed things about whether this should be suppressed with `role=”presentation”` vs whether it is useful so just flagging here for @jenhadfield-dfe to consider!

File: app/components/footer_component.html.erb [hunk 3]
```diff
@@ -1,35 +1,98 @@
...
+
+      <div class="site-footer__links-container">
+        <div class="site-footer__links-column">
+          <h3 class="heading-m">Become a teacher</h3>
```
Comment (gemmadallmandfe) [inline app/components/footer_component.html.erb:39]: @Sarah-DfE I think this should be an h2 - it doesn't really sit under the h2 of Talk to us - would you agree?

If so, the `Finance and funding` and `Routes into teaching` h3s would also need changing to h2s

File: app/components/footer_component.html.erb [hunk 4]
```diff
@@ -1,35 +1,98 @@
...
+            <li><a class="link--black" href="/train-to-be-a-teacher/teacher-degree-apprenticeships">Teacher degree apprenticeship</a></li>
+          </ul>
+        </div>
+      </div>
+
+      <hr class="site-footer__divider">
+    </section>
+
+    <div class="site-footer__social-icons">
+      <a href="https://www.facebook.com/getintoteaching" class="site-footer-top__social__link" target="_blank" rel="noopener">
```
Comment (gemmadallmandfe) [inline app/components/footer_component.html.erb:74]: Social media icons disappear in active state (black on black) @jenhadfield-dfe 

image

**Review by gemmadallmandfe — COMMENTED** [review]
Looks great! A few comments re accessibility for you (will have a look at how it works around the site next)

**Review by martyn-w — COMMENTED** [review]
👍 LGTM - one some small typo to fix

File: app/components/footer_component.html.erb [hunk 5]
```diff
@@ -1,35 +1,98 @@
...
+          </ul>
+        </div>
+        <div class="site-footer__links-column">
+          <h3 class="heading-m">Routes into teaching</h3>
+          <ul>
+            <li><a class="link--black" href="/train-to-be-a-teacher/if-you-have-a-degree">With a degree</a></li>
+            <li><a class="link--black" href="/train-to-be-a-teacher/if-you-dont-have-a-degree">Without a degree</a></li>
+            <li><a class="link--black" href="/train-to-be-a-teacher/assessment-only-route-to-qts">Assessment only QTS</a></li>
+            <li><a class="link--black" href="/life-as-a-teacher/teaching-as-a-career/change-careers">Retrain as a teacher</a></li>
+            <li><a class="link--black" href="/train-to-be-a-teacher/teacher-degree-apprenticeships">Teacher degree apprenticeship</a></li>
```
[the remark this answers is not in the document]
Comment (Sarah-DfE) [reply app/components/footer_component.html.erb:65]: @gemmadallmandfe agree, makes sense!

File: app/components/footer_component.html.erb [hunk 6]
```diff
@@ -41,7 +104,7 @@
         <div class="site-footer-bottom__links-container__license">All content is available under the <a href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/">Open Government Licence v3.0,</a> except where otherwise stated.</div>
       </div>
       <div class="site-footer-bottom__logo-container">
-        <%= image_pack_tag 'static/images/dfelogo-white.svg', alt: "Department for Education logo", size: "124x73" %>
+        <%= image_pack_tag 'static/images/dfelogo-black.svg', alt: "Department for Education logo", size: "124x73" %>
```
[the remark this answers is not in the document]
Comment (gemmadallmandfe) [reply app/components/footer_component.html.erb:107]: However, if we do add in a link, presumably we would need the reverse colour logo for the active state when keyboard users tab to it?

File: app/components/footer_component.html.erb [hunk 7]
```diff
@@ -1,35 +1,98 @@
...
+            <li><a class="link--black" href="/life-as-a-teacher/teaching-as-a-career/change-careers">Retrain as a teacher</a></li>
+            <li><a class="link--black" href="/train-to-be-a-teacher/teacher-degree-apprenticeships">Teacher degree apprenticeship</a></li>
+          </ul>
+        </div>
+      </div>
+
+      <hr class="site-footer__divider">
+    </section>
+
+    <div class="site-footer__social-icons">
```
[the remark this answers is not in the document]
Comment (Sarah-DfE) [reply app/components/footer_component.html.erb:73]: I say we go with 'Support links' if that's what GOV.UK do!

I've had a look at a few other sites and seems common to have these extra links at the bottom without a visible heading.

File: app/views/content/landing/campus-mailing-list.md [hunk 8]
```diff
@@ -7,7 +7,6 @@ content:
 image: "static/images/content/hero-images/chemistry.jpg"
 colour: pink-yellow
 layout: "layouts/minimal"
-talk_to_us: false
```
Comment (gemmadallmandfe) [inline app/views/content/landing/campus-mailing-list.md:10]: On the `/landing/campus-advisers` page, there is some breathing space between the body and the footer on mobile (albeit that it looks a bit stripy!)

However, on the `/landing/campus-mailing-list` page, there isn't the same kind of separation, which looks very odd?

@jenhadfield-dfe @spencerldixon 

image

image

File: app/views/content/landing/how-much-do-teachers-get-paid.md [hunk 9]
```diff
@@ -10,7 +10,6 @@ content:
 image: "static/images/content/hero-images/science.jpg"
 colour: "pink-blue"
 layout: "layouts/minimal"
-talk_to_us: false
```
Comment (gemmadallmandfe) [inline app/views/content/landing/how-much-do-teachers-get-paid.md:13]: Related to the comments on the `/landing/campus-mailing-list` page, does there need to be a little more padding above the footer generally @jenhadfield-dfe ?

image

File: app/views/content/landing/mailing-list-1.md [hunk 10]
```diff
@@ -6,7 +6,6 @@ content:
     - content/landing/mailing-list-1/mailing-list
 image: "static/images/content/hero-images/english.jpg"
 layout: "layouts/minimal"
-talk_to_us: false
```
Comment (gemmadallmandfe) [inline app/views/content/landing/mailing-list-1.md:9]: Same issue re lack of gap above footer
image

File: app/views/content/landing/mailing-list-2.md [hunk 11]
```diff
@@ -6,7 +6,6 @@ content:
     - content/landing/mailing-list-2/mailing-list
 colour: pink-blue
 layout: "layouts/minimal"
-talk_to_us: false
```
Comment (gemmadallmandfe) [inline app/views/content/landing/mailing-list-2.md:9]: Same issue with lack of gap above footer

File: app/views/content/landing/mailing-list-3.md [hunk 12]
```diff
@@ -6,7 +6,6 @@ content:
     - content/landing/mailing-list-3/mailing-list
 image: "static/images/content/hero-images/english.jpg"
 layout: "layouts/minimal"
-talk_to_us: false
```
Comment (gemmadallmandfe) [inline app/views/content/landing/mailing-list-3.md:9]: Same issue with lack of gap above footer

File: app/views/content/landing/mailing-list.md [hunk 13]
```diff
@@ -6,7 +6,6 @@ content:
     - content/landing/mailing-list/mailing-list
 image: "static/images/content/hero-images/english.jpg"
 layout: "layouts/minimal"
-talk_to_us: false
```
Comment (gemmadallmandfe) [inline app/views/content/landing/mailing-list.md:9]: Same issue with lack of gap above footer

**Review by gemmadallmandfe — COMMENTED** [review]
@spencerldixon @jenhadfield-dfe I've had a look at other pages around the site and I think we should probably have a look at how the footer sits below different content - it is very close in a lot of cases and sometimes doesn't look very separate if the content above is also on a grey background

NB I've commented multiple times but it mostly comes down to the same thing, I think!

File: app/views/content/landing/campus-mailing-list.md (hunk 8 above)
Comment (spencerldixon) [reply app/views/content/landing/campus-mailing-list.md:10]: @gemmadallmandfe I've added a `margin-top` to the footer to keep a bit of space between the main content and the footer, let me know what you think

Comment (gemmadallmandfe) [reply app/views/content/landing/campus-mailing-list.md:10]: That looks better to me! @jenhadfield-dfe are you happy with this?

Comment (jenhadfield-dfe) [reply app/views/content/landing/campus-mailing-list.md:10]: yes looks much better!

File: app/components/footer_component.html.erb (hunk 7 above)
[the remark this answers is not in the document]
Comment (gemmadallmandfe) [reply app/components/footer_component.html.erb:73]: @spencerldixon Please can you add in a visually hidden h2 of `Support links` after the `hr` and before the social media icons? I will leave to you as I don't know whether it needs its own `section`!

File: app/components/footer_component.html.erb (hunk 6 above)
[the remark this answers is not in the document]
Comment (gemmadallmandfe) [reply app/components/footer_component.html.erb:107]: @jenhadfield-dfe In the interests of getting this PR out soon, I'm happy to raise a separate ticket to look at this on your return (if you think we should do this!)

File: app/components/footer_component.html.erb (hunk 2 above)
Comment (gemmadallmandfe) [reply app/components/footer_component.html.erb:35]: Resolving this - we are replicating GOV.UK so let's leave as is

File: app/components/footer_component.html.erb (hunk 4 above)
Comment (gemmadallmandfe) [reply app/components/footer_component.html.erb:74]: @spencerldixon Would you be able to add the reverse social media icons for the active state, so they are visible when tabbed to? We have the white versions already in production on the black footer

File: app/components/footer_component.html.erb (hunk 7 above)
[the remark this answers is not in the document]
Comment (spencerldixon) [reply app/components/footer_component.html.erb:73]: Done!

File: app/components/footer_component.html.erb (hunk 4 above)
Comment (spencerldixon) [reply app/components/footer_component.html.erb:74]: Done!

File: app/components/footer_component.html.erb (hunk 6 above)
[the remark this answers is not in the document]
Comment (spencerldixon) [reply app/components/footer_component.html.erb]: Done!

File: app/components/footer_component.html.erb [hunk 14]
```diff
@@ -1,37 +1,101 @@
...
+      </a>
+      <a href="http://www.youtube.com/user/getintoteaching" class="site-footer-top__social__link" target="_blank" rel="noopener">
+        <%= helpers.fab_icon("youtube", "icon") %>
+        <span class="visually-hidden">Youtube - opens in a new tab</span>
+      </a>
     </div>
+
     <div class="site-footer-bottom">
       <div class="site-footer-bottom__links-container">
+        <h2 class="visually-hidden">Support links</h2>
```
Comment (gemmadallmandfe) [inline app/components/footer_component.html.erb:98]: @spencerldixon This works really well but I think we need to move it above the social media icons (around line 74?) - could you possibly do that please?

File: app/components/footer_component.html.erb [hunk 15]
```diff
@@ -41,7 +105,9 @@
         <div class="site-footer-bottom__links-container__license">All content is available under the <a href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/">Open Government Licence v3.0,</a> except where otherwise stated.</div>
       </div>
       <div class="site-footer-bottom__logo-container">
-        <%= image_pack_tag 'static/images/dfelogo-white.svg', alt: "Department for Education logo", size: "124x73" %>
+        <%= link_to root_path do %>
```
Comment (gemmadallmandfe) [inline app/components/footer_component.html.erb:108]: @spencerldixon Link needs to go to the DfE home page (not ours!) which is `https://www.gov.uk/government/organisations/department-for-education`

File: app/components/footer_component.html.erb [hunk 16]
```diff
@@ -41,7 +105,9 @@
...
+          <%= image_pack_tag 'static/images/dfelogo-black.svg', alt: "Department for Education logo", size: "124x73" %>
```
Comment (gemmadallmandfe) [inline app/components/footer_component.html.erb:109]: @spencerldixon This isn't quite working - the logo is not visible when tabbed to using keyboard only

image

File: app/components/footer_component.html.erb (hunk 4 above)
Comment (gemmadallmandfe) [reply app/components/footer_component.html.erb:74]: @spencerldixon This isn't quite working still - icons are still not visible when tabbed to using keyboard only:

image

File: app/components/footer_component.html.erb (hunk 16 above)
Comment (gemmadallmandfe) [reply app/components/footer_component.html.erb:109]: Logo looks good now - resolving this - thanks!

File: app/components/footer_component.html.erb (hunk 4 above)
Comment (gemmadallmandfe) [reply app/components/footer_component.html.erb:75]: Icons look good now - resolving this - thanks!

**Review by gemmadallmandfe — DISMISSED** [review]
Looks good to me

Happy for this to be merged as long as @martyn-w is happy that Chat is functioning correctly (looks OK from the front end)

**Review by martyn-w — DISMISSED** [review]
👍 LGTM

**Review by martyn-w — APPROVED** [review]
👍 LGTM

**Pull request merged**